File: node/challenge4.js
//Implement reduceLeft

function reduceLeft(xs, z, f) {
    for (let i = 0; i < xs.length; i++) {
        z = f(z, xs[i])
    }
    return z
}
const lst = [1, 2, 3, 4, 5]
const res = reduceLeft(lst, 0, (x, y) => x + y)
console.log(res)
  

File: python/challenge4.py
#Implement reduceLeft using reduce.

from typing import List
from functools import reduce

def reduce_left(xs: List, z, f):
    return reduce(f, xs, z)

lst = [1, 2, 3]
res = reduce_left(lst, 0, lambda x, y: x + y)
print(res)
